src/org/loklak/harvester/TwitterScraper.java

```diff
@@ -520,8 +520,8 @@ public static String unshorten(String text) {
             try {
                 Matcher m = timeline_embed_pattern.matcher(text);
                 if (m.find()) {
-                    String shorturl = RedirectUnshortener.unShorten(m.group(2));
-                    text = m.replaceFirst("https://pic.twitter.com/" + shorturl + " ");
+                    String shorturl = RedirectUnshortener.unShorten(m.group(1));
+                    text = m.replaceFirst(" https://pic.twitter.com/" + shorturl + " ");

```

Comment:
Is it needed to `replaceFirst()` with the initial ` ` characters ? Would it not be possible to trim the replacement when Matching the required pattern with the matcher ?